File: framework/web/src/main/java/cn/bestwu/simpleframework/support/client/ClientHttpResponseWrapper.java
package cn.bestwu.simpleframework.support.client;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.client.ClientHttpResponse;
import org.springframework.util.StreamUtils;

/**
 * 响应结果包装类 可反复取响应内容
 *
 * @author Peter Wu
 */
public class ClientHttpResponseWrapper implements ClientHttpResponse {

  private final ClientHttpResponse response;
  private final byte[] bytes;

  public ClientHttpResponseWrapper(ClientHttpResponse response) throws IOException {
    this.response = response;
    bytes = StreamUtils.copyToByteArray(response.getBody());
  }

  public String getRecord() {
    return new String(bytes);
  }

  @Override
  public HttpHeaders getHeaders() {
    return this.response.getHeaders();
  }

  @Override
  public InputStream getBody() throws IOException {
    return new ByteArrayInputStream(bytes);
  }


  @Override
  public HttpStatus getStatusCode() throws IOException {
    return this.response.getStatusCode();
  }

  @Override
  public int getRawStatusCode() throws IOException {
    return this.response.getRawStatusCode();
  }

  @Override
  public String getStatusText() throws IOException {
    return this.response.getStatusText();
  }

  @Override
  public void close() {
    this.response.close();
  }

}

File: framework/starter-logging/src/main/kotlin/cn/bestwu/logging/client/ClientHttpResponseWrapper.kt
package cn.bestwu.logging.client

import org.springframework.http.HttpHeaders
import org.springframework.http.HttpStatus
import org.springframework.http.client.ClientHttpResponse
import org.springframework.util.StreamUtils
import java.io.IOException
import java.io.ByteArrayInputStream
import java.io.InputStream

/**
 * 响应结果包装类 可反复取响应内容
 *
 * @author Peter Wu
 */
class ClientHttpResponseWrapper(private val response: ClientHttpResponse?) : ClientHttpResponse {

    val bytes: ByteArray =
        if (response != null) StreamUtils.copyToByteArray(response.body) else ByteArray(0)

    override fun getHeaders(): HttpHeaders {
        return response?.headers ?: HttpHeaders.EMPTY
    }

    @Throws(IOException::class)
    override fun getBody(): InputStream {
        return ByteArrayInputStream(bytes)
    }

    @Throws(IOException::class)
    override fun getStatusCode(): HttpStatus {
        return response?.statusCode ?: HttpStatus.INTERNAL_SERVER_ERROR
    }

    @Throws(IOException::class)
    override fun getRawStatusCode(): Int {
        return response?.rawStatusCode ?: HttpStatus.INTERNAL_SERVER_ERROR.value()
    }

    @Throws(IOException::class)
    override fun getStatusText(): String {
        return response?.statusText ?: ""
    }

    override fun close() {
        response?.close()
    }

}